colossus-docs/src/main/scala/GenericClient.scala

```diff
@@ -0,0 +1,50 @@
+import akka.actor.ActorSystem
+import colossus.IOSystem
+import colossus.protocols.http.HttpMethod.Get
+import colossus.protocols.http.UrlParsing.{Root, on}
+import colossus.protocols.http.server.{HttpServer, Initializer, RequestHandler}
+import colossus.protocols.http.{Http, HttpClient, HttpRequest}
+import colossus.service.GenRequestHandler.PartialHandler
+import colossus.service.{Async, CallbackAsync, FutureAsync}
+
+import scala.language.higherKinds
+
+object GenericClient extends App {
+
+  // #example
+  implicit val actorSystem = ActorSystem()
+  implicit val ioSystem = IOSystem()
+
+  def doTheThing[A[_]](client: HttpClient[A])(implicit async: Async[A]): A[String] = {
+    async.map(client.send(HttpRequest.get("/"))) { response =>
+      response.body.bytes.utf8String
+    }
+  }
+
+  // using future client
+  implicit val futureAsync = new FutureAsync()
```

Comment:
Hmm, you shouldn't have to be creating your own `FutureAsync`.  Well, at least that was my intention.  This stuff is currently used in [ClientFactory](https://github.com/tumblr/colossus/blob/master/colossus/src/main/scala/colossus/service/ServiceDSL.scala#L226), and notice that an `AsyncBuilder` is used.  

This is probably out of scope for this PR, but it should ideally be the case that all you need to do is import a sub-package or object containing some implicits, and let the builder do the work for you.